packages/flutter_tools/test/integration/flutter_run_integration_test.dart

```diff
@@ -0,0 +1,171 @@
+    return _flutter.breakAt(
+      fs.path.join(_tempDir.path, 'lib', 'main.dart'),
+      9
+    );
+  }
+
+  Future<VMIsolate> breakInTopLevelFunction(FlutterTestDriver flutter) async {
+    return _flutter.breakAt(
+      fs.path.join(_tempDir.path, 'lib', 'main.dart'),
+      17
+    );
+  }
+
+  group('FlutterTesterDevice', () {
+
+    testUsingContext('can hot reload', () async {
+      await _flutter.run();
+      await _flutter.hotReload();
+    }, skip: true); // https://github.com/flutter/flutter/issues/17833
+
+    testUsingContext('can hit breakpoints with file:// prefixes after reload', () async {
+      await _flutter.run(withDebugger: true);
+      
+      // Add the breakpoint using a file:// URI.
+      await _flutter.addBreakpoint(
+          // Test currently passes with a FS path, but not with file:// URI.
+          // fs.path.join(_tempDir.path, 'lib', 'main.dart'),
+          new Uri.file(fs.path.join(_tempDir.path, 'lib', 'main.dart')).toString(),
+          9
+      );
+      
+      await _flutter.hotReload();
+
+      // Ensure we hit the breakpoint.
+      final VMIsolate isolate = await _flutter.waitForBreakpointHit();
+      expect(isolate.pauseEvent, const isInstanceOf<VMPauseBreakpointEvent>());
+    }, skip: true); // https://github.com/flutter/flutter/issues/18441
+
+    Future<void> evaluateTrivialExpressions() async {
+      VMInstanceRef res;
+
+      res = await _flutter.evaluateExpression('"test"');
+      expect(res is VMStringInstanceRef && res.value == 'test', isTrue);
+
+      res = await _flutter.evaluateExpression('"test"');
+      expect(res is VMIntInstanceRef && res.value == 1, isTrue);
+
+      res = await _flutter.evaluateExpression('"test"');
+      expect(res is VMBoolInstanceRef && res.value == true, isTrue);
+    }
+
+    Future<void> evaluateComplexExpressions() async {
+      final VMInstanceRef res = await _flutter.evaluateExpression('new DateTime.now().year');
+      expect(res is VMIntInstanceRef && res.value == new DateTime.now().year, isTrue);
+    }
+
+    Future<void> evaluateComplexReturningExpressions() async {
+      final DateTime now = new DateTime.now();
+      final VMInstanceRef resp = await _flutter.evaluateExpression('new DateTime.now()');
+      expect(resp.klass.name, equals('DateTime'));
+      final DateTime value = await resp.getValue();
+      // Ensure we got a reasonable approximation. The more accurate we try to
+      // make this, the more likely it'll fail due to differences in the time
+      // in the remote VM and the local VM.
+      expect('${value.year}-${value.month}-${value.day}',
+          equals('${now.year}-${now.month}-${now.day}'));
+    }
+
+    testUsingContext('can evaluate trivial expressions in top level function', () async {
+      await _flutter.run(withDebugger: true);
+      await breakInTopLevelFunction(_flutter);
+      await evaluateTrivialExpressions();
+    }, skip: true); // https://github.com/flutter/flutter/issues/18678
```

Comment:
For some (many?) of these skips, we should instead invert the test to expect failure. That means we'll be notifies by the CI when the issue is fixed.